test/ScriptCs.Hosting.Tests/ObjectSerializerTests.cs

```diff
@@ -16,6 +16,34 @@ public TheSerializeMethod()
             }
 
             [Fact]
+            public void ShouldSerialize()
+            {
+                // arrange
+                var obj = new Foo
+                {
+                    Bar = new Bar
+                    {
+                        Baz = true,
+                        Bazz = 123.4,
+                        Bazzz = "hello",
+                    },
+                };
+
+                // act
+                var result = _serializer.Serialize(obj);
+
+                // assert
+                result.ShouldEqual(
+@"{
+  ""Bar"": {
+    ""Baz"": true,
+    ""Bazz"": 123.4,
+    ""Bazzz"": ""hello""
+  }
+}");
+            }
+
+            [Fact]
             public void ShouldSerializeTypeMethods()
```

Comment:
Would be nice to change this test and ShouldSerializeDelegates to verify the result similar to what you did in ShouldSerialize. All it does now is verify that there is no exception, not that the result is correct ;-)